modules/stringtie/1.0/stringtie.smk

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Krysta Coyle
+# Module Author:    Krysta Coyle
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["stringtie"]`
+CFG = op.setup_module(
+    name = "stringtie",
+    version = "1.0",
+    subdirectories = ["inputs", "stringtie", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _stringtie_input_bam,
+    _stringtie_output_gtf,
+    _stringtie_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _stringtie_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    group: 
```

Comment:
Thank you Krysta, very good module!!! I would chime into the review suggesting to drop the grouping here since it is not used in the module.

Reply:
I actually would recommend keeping the grouping but adding it to the _stringtie_run rule - it's meant to group the fast-running input job with the longer-running job to prevent overwhelming the head node or gphost with more jobs than allowed per user.

Reply:
yes, sure, thanks Laura!

Reply:
Moved the grouping to the _stringtie_run rule